restli-server/src/main/java/com/linkedin/restli/internal/server/model/RestLiAnnotationReader.java

```diff
@@ -554,6 +554,20 @@ else if (KeyUnstructuredDataResource.class.isAssignableFrom(collectionResourceCl
         @SuppressWarnings("unchecked")
         Class<? extends KeyUnstructuredDataResource<?>> clazz = (Class<? extends KeyUnstructuredDataResource<?>>) collectionResourceClass;
         actualTypeArguments = ReflectionUtils.getTypeArgumentsParametrized(KeyUnstructuredDataResource.class, clazz);
+        for (Method method : collectionResourceClass.getDeclaredMethods())
+        {
+          if (method.isSynthetic())
+          {
+            continue;
+          }
+          Finder finderAnno = method.getAnnotation(Finder.class);
+          if (finderAnno == null)
+          {
+            continue;
+          }
+          throw new ResourceConfigException("Class '" + collectionResourceClass.getSimpleName() + "' of a"
```

Comment:
nit: `Class X of a Y class` doesn't really make sense. It would be more clear to say `Class X extending Y`

Also, I'd suggest not hard-coding the class names into the String message. You should programmatically get the class name. This lets the navigate/refactor tool find this reference.

Reply:
Done